README.md

```diff
@@ -37,3 +37,175 @@ docker run -it --rm \
+         libudev-dev \
+         libssl-dev && \
+       rm -rf /var/lib/apt/lists/*
+
+   WORKDIR /usr/src/boardswarm
+
+   RUN cargo build \
+         --release
+   ```
+   Save this to the root of the git repos once we've checked it out below.
+
+If not building from a container (where this'll be handled by the container
+creation step), install required tools/libraries to build boardswarm. As we'll
+be pulling the latest from git, we'll need that installed too. Again, assuming
+a Debian based system:
+```
+$ sudo apt install rustc git pkg-config protobuf-compiler libudev-dev libssl-dev
+```
+
+Boardswarm is quite new and undergoing active development, so as previously
+mentioned, we'll build the latest from git. Clone the boardswarm repository:
+```
+$ git clone https://github.com/boardswarm/boardswarm.git
+```
+
+We can now build boardswarm:
+```
+$ cd boardswarm
+$ cargo build --release
+```
+Or, if wanting to use a container for this, run the following command:
+```
+podman build --volume=$(pwd):/usr/src/boardswarm .
+```
+
+### Server Install
+
+The "server" is the machine that will have the development boards attached to
+it.
+
+1. Copy the `boardswarm` binary to `/usr/local/bin/` on the server.
+
+1. Create a boardswarm user and group for the boardswarm server to run as:
+   ```
+   $ sudo addgroup --system boardswarm
+   $ sudo adduser --system --ingroup boardswarm --no-create-home --disabled-password boardswarm
+   ```
+
+1. Copy the [example systemd unit](boardswarm/share/boardswarm.service) to
+   `/etc/systemd/system/boardswarm.service`.
+
+1. Copy [example udev rules](boardswarm/share/99-boardswarm.rules) to
+   `/etc/udev/rules.d/`.
+
+1. Reload udev rules and ensure they are applied:
+   ```
+   $ sudo udevadm control --reload-rules
+   $ sudo udevadm trigger
+   ```
+
+1. Copy example [`server.conf`](boardswarm/share/server.conf) to
+   `/etc/boardswarm/server.conf` on target server machine and modify as
+   required.
+
+### Server Configuration
+
+1. A mechanism for controlling power to the devices is generally required when
+   working with boards remotely. Whilst in some instances this might be handled
+   using gpio actuators, boardswarm has support for
+   [pdudaemon](https://github.com/pdudaemon/pdudaemon). Pdudaemon provides
+   support for a wide range of programmable power supplies and its use if
+   encouraged.  If using it for power mangement install and configure it to
+   switch the development boards on and off, using whichever mechanism you are
+   using (ensure that the `server.conf` `pdu` provider is configured to match the
+   configuration set in the pdudaemon configuration):
+   ```
+   $ sudo apt install pdudaemon
+   ```
+
+1. We are going to use Static JWKS based authentication, as documented in the
+   [server README](boardswarm/README.md#static-jwks-based-authentication).
+   Install `rnbyc` so we can create the required tokens:
+   ```
+   $ sudo apt install rnbyc
+   ```
+
+1. Generate an EdDSA key:
```

Comment:
rather not have these instructions in two places; they're bound to get out of sync